### PR TITLE
commit for fix NonUniqueObjectException

### PR DESCRIPTION
**Pull request by MaxFedosenko**

### CODE REVIEW

File: src/main/java/com/tms/web/config/HibernateConfiguration.java [hunk 1]
```diff
@@ -24,10 +26,34 @@ public static void prepareHibernate() {
...
+        configuration.addAnnotatedClass(Client.class);
+        configuration.addAnnotatedClass(Region.class);
 //        configuration.addAnnotatedClass(Parent.class);
-        EntityScanner.scanPackages("com.tms.web.entity").addTo(configuration);
+//        configuration.addAnnotatedClass(Home.class);
 
         sessionFactory = configuration.buildSessionFactory();
     }
+
+//    public static void testParent(){
```
Comment (gr00shik) [inline src/main/java/com/tms/web/config/HibernateConfiguration.java:38]: please don't pull comments

File: src/main/java/com/tms/web/service/ClientService.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package com.tms.web.service;
+
+import com.tms.web.entities.Car;
+import com.tms.web.entities.Client;
+
+import java.util.List;
+
+public interface ClientService {
+
+    void saveClient(String name);
```
Comment (gr00shik) [inline src/main/java/com/tms/web/service/ClientService.java:10]: just save

File: src/main/java/com/tms/web/service/RegionService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.tms.web.service;
+
+import com.tms.web.entities.Car;
+
+public interface RegionService {
+
+    void addRegion (String name);
```
Comment (gr00shik) [inline src/main/java/com/tms/web/service/RegionService.java:7]: save or add

File: src/main/java/com/tms/web/serviceImpl/CarServiceImpl.java [hunk 4]
```diff
@@ -39,14 +45,14 @@ public void save(Car car) {
     }
 
     @Override
-    public void getByID(Long id) {
+    public Car getByID(Long id) {
         Session session = sessionFactory.openSession();
         Transaction transaction = session.beginTransaction();
         Car car = session.createQuery("from Car where id = :id", Car.class)
```
Comment (gr00shik) [inline src/main/java/com/tms/web/serviceImpl/CarServiceImpl.java:51]: you can use get to request object by id

File: src/main/java/com/tms/web/serviceImpl/ClientServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        session.close();
+    }
+
+    @Override
+    public void addCar(Car car, Long id) {
+        Session session = sessionFactory.openSession();
+        Transaction transaction = session.beginTransaction();
+        Client client = session.createQuery("from Client where id = :id", Client.class)
+                .setParameter("id", id).getSingleResult();
+        car.setClient(client);
```
Comment (gr00shik) [inline src/main/java/com/tms/web/serviceImpl/ClientServiceImpl.java:32]: change the service as we discussed

File: src/main/java/com/tms/web/serviceImpl/RegionServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+public class RegionServiceImpl implements RegionService {
+    @Override
+    public void addCars(Car car, Long id) {
+        Session session = sessionFactory.openSession();
+        Transaction transaction = session.beginTransaction();
+        Region region = session.createQuery("from Region where id = :id", Region.class)
+                .setParameter("id", id).getSingleResult();
+        car.getRegions().add(region);
+        region.getCars().add(car);
+        session.saveOrUpdate(region);
```
Comment (gr00shik) [inline src/main/java/com/tms/web/serviceImpl/RegionServiceImpl.java:20]: the same

File: src/main/java/com/tms/web/servlet/AddCars.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+@WebServlet("/addcars")
+public class AddCars extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Long idCar = Long.valueOf(req.getParameter("idCar"));
+        Long idClient = Long.valueOf(req.getParameter("idClient"));
+        ClientService clientService = new ClientServiceImpl();
+        CarService carService = new CarServiceImpl();
+        Car car = carService.getByID(idCar);
```
Comment (gr00shik) [inline src/main/java/com/tms/web/servlet/AddCars.java:25]: don't use a lot of logic in presentation layer

File: src/main/java/com/tms/web/servlet/AddRegion.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+@WebServlet("/addregion")
+public class AddRegion extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        Long idCar = Long.valueOf(req.getParameter("idCar"));
+        Long idRegion = Long.valueOf(req.getParameter("idRegion"));
+        RegionService regionService = new RegionServiceImpl();
+        CarService carService = new CarServiceImpl();
+        Car car = carService.getByID(idCar);
```
Comment (gr00shik) [inline src/main/java/com/tms/web/servlet/AddRegion.java:27]: the same

File: src/main/java/com/tms/web/servlet/SaveClient.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+@WebServlet("/client")
+public class SaveClient extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String name = req.getParameter("name");
+        ClientService clientService = new ClientServiceImpl();
+        clientService.saveClient(name);
+
+
```
Comment (gr00shik) [inline src/main/java/com/tms/web/servlet/SaveClient.java:32]: extra lines

**Review by gr00shik — COMMENTED** [review]
ok

File: src/main/java/com/tms/web/config/HibernateConfiguration.java (hunk 1 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/config/HibernateConfiguration.java:38]: fix

File: src/main/java/com/tms/web/service/ClientService.java (hunk 2 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/service/ClientService.java:10]: fix

File: src/main/java/com/tms/web/service/RegionService.java (hunk 3 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/service/RegionService.java:7]: fix

File: src/main/java/com/tms/web/serviceImpl/CarServiceImpl.java (hunk 4 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/serviceImpl/CarServiceImpl.java:51]: fix

File: src/main/java/com/tms/web/serviceImpl/ClientServiceImpl.java (hunk 5 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/serviceImpl/ClientServiceImpl.java:32]: fix

File: src/main/java/com/tms/web/serviceImpl/RegionServiceImpl.java (hunk 6 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/serviceImpl/RegionServiceImpl.java:20]: fix

File: src/main/java/com/tms/web/servlet/AddCars.java (hunk 7 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/servlet/AddCars.java:25]: fix

File: src/main/java/com/tms/web/servlet/AddRegion.java (hunk 8 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/servlet/AddRegion.java:27]: fix

File: src/main/java/com/tms/web/servlet/SaveClient.java (hunk 9 above)
Comment (MaxFedosenko) [reply src/main/java/com/tms/web/servlet/SaveClient.java:32]: fix

**Pull request merged**